pkg/httpclient/settings.go

```diff
@@ -64,6 +69,7 @@ func (s *Settings) Validate() error {
 func Configure(settings *Settings) {
 	transport := http.DefaultTransport.(*http.Transport)
 
+	http.DefaultClient.Timeout = settings.Timeout
```

Comment:
we use the default transport in some places in the code (without the default client).  Have you checked if what we configured on it is sufficient ?